src/RunnableQueue.cpp

```diff
@@ -21,52 +21,52 @@ namespace vrb {
 struct RunnableQueue::State {
   Mutex lock;
   JavaVM* vm;
-  JNIEnv* processEnv;
-  jclass runnableClass;
+  JNIEnv* threadEnv;
   jmethodID runMethod;
   std::vector<jobject> list[2];
   uint32_t count;
   State()
       : vm(nullptr)
-      , processEnv(nullptr)
-      , runnableClass(nullptr)
+      , threadEnv(nullptr)
       , runMethod(nullptr)
       , count(0)
   {}
   ~State() {
-    if (!vm) { return; }
-    JNIEnv* env = nullptr;
-    if (!vm->AttachCurrentThread(&env, nullptr)) { return; }
-    if (runnableClass) {
-      env->DeleteGlobalRef(runnableClass);
-    }
-    for (int ix = 0; ix < 2; ix++) {
-      for (jobject& runnable: list[ix]) {
-        env->DeleteGlobalRef(runnable);
-        runnable = nullptr;
-      }
-      list[ix].clear();
-    }
+    Shutdown();
   }
-  void Init(JNIEnv* aEnv) {
-
+  void Initialize(JNIEnv* aEnv) {
     jclass localRunnableClass = aEnv->FindClass("java/lang/Runnable");
     if (!localRunnableClass) {
       VRB_ERROR("Failed to fine java class java/lang/Runnable in %s", __FILE__);
       return;
     }
-    runnableClass = (jclass)aEnv->NewGlobalRef(localRunnableClass);
+    jclass runnableClass = (jclass)aEnv->NewGlobalRef(localRunnableClass);
```

Comment:
I don't think we need to make the localRunnableClass global here, just call DeleteLocalRef

Reply:
fixed